app/src/format.py

```diff
@@ -102,19 +102,41 @@ def _format_to_accordion_html(document: Document, score: float) -> str:
 def _format_to_accordion_group_html(documents: OrderedDict[Document, list[ChunkWithScore]]) -> str:
     global _accordion_id
     html = ""
+    citation_number = 1
     for document in documents:
-        internal_citation = ""
+        citations = ""
         _accordion_id += 1
-        for index, chunk in enumerate(documents[document], start=1):
-            chunk_lines = chunk.chunk.content.splitlines()
-            formatted_chunk = " ".join(chunk_lines)
-            formatted_chunk = re.sub(r"\t+", "", formatted_chunk).strip()
-            formatted_chunk = (
-                f"<p>{_replace_bem_with_link(formatted_chunk)} </p>" if formatted_chunk else ""
+
+        citation_number_start = citation_number
+
+        for chunk_with_score in documents[document]:
+            chunk = chunk_with_score.chunk
+            formatted_chunk = _replace_bem_with_link(chunk.content)
+
+            # Adjust markdown for lists so Chainlit renders correctly
+            formatted_chunk = formatted_chunk.replace("    - ", "- ")
+            if formatted_chunk.startswith("- "):
+                formatted_chunk = "\n" + formatted_chunk
```

Comment:
Note to self: Currently, this can happen because the previous sentence doesn't end with a colon. It can also happen because the list is the first element after a new heading.